### PR TITLE
feat: creating words backup screen

### PR DESCRIPTION
**Pull request by pedroferreira1**
image
image
image

Closes https://github.com/HathorNetwork/hathor-wallet-mobile/issues/23

### CODE REVIEW

File: src/screens/BackupWords.js [hunk 1]
```diff
@@ -0,0 +1,241 @@
...
+  constructor(props) {
+    super(props);
+    const paramWords = this.props.navigation.getParam('words');
+    this.words = paramWords ? paramWords.split(' ') : [];
+  }
+
+  componentDidMount() {
+    // Get 5 random indexes of word to check backup
+    const indexesArr = Array(24).fill().map((v, i) => i + 1);
+    const indexesToBackup = _.shuffle(indexesArr).slice(0, 5);
```
Comment (obiyankenobi) [inline src/screens/BackupWords.js]: With this shuffle, we can request the words out of order (eg: 2, 9, 5, 22, 15). I actually don't mind, but usually I think other devices request them in order (eg: 2, 5, 9, 15, 22).

Comment (pedroferreira1) [reply src/screens/BackupWords.js]: My idea was exactly to show the words in random order, I thought it was better. We can discuss this, it's easy to order the 5 indexes after the shuffle but at first I wanted not to be in order.

File: src/screens/BackupWords.js [hunk 2]
```diff
@@ -0,0 +1,241 @@
...
+   */
+  updateWordOptions = () => {
+    const index = this.state.indexes[this.state.step] - 1;
+    let optionsStartIndex = index - 2;
+    let optionsEndIndex = index + 2;
+
+    // If index is 0 or 1, startIndex would be negative
+    // So we set to 0 and increment the endIndex
+    if (optionsStartIndex < 0) {
+      optionsEndIndex += -optionsStartIndex;
```
Comment (msbrogli) [inline src/screens/BackupWords.js]: Isn't it better to set to 4?

File: src/screens/BackupWords.js [hunk 3]
```diff
@@ -0,0 +1,241 @@
...
+      optionsStartIndex = 0;
+    }
+
+    // If index is 22 or 23, endIndex would be greater than the max
+    // So we set to the max and decrease the startIndex
+    const maxIndex = this.words.length - 1;
+    if (optionsEndIndex > maxIndex) {
+      optionsStartIndex -= optionsEndIndex - maxIndex;
```
Comment (msbrogli) [inline src/screens/BackupWords.js]: Isn't it better to set to `maxIndex - 4`?

File: src/screens/BackupWords.js [hunk 4]
```diff
@@ -0,0 +1,241 @@
...
+
+  /**
+   * Update state with options to be shown to the user depending on the step
+   * If the step is the word in position 4, we must show the words in position 2, 3, 4, 5, 6
+   * If we get one of the corner cases
+   * (positions 1, 2, 23, or 24 we expand to the side we still have words) e.g., if we want to
+   * validate word in position 2, we must have 5 options.
+   * Positions 1, 2, 3, 4, 5 (the position 0 is substituted by position 5)
+   */
+  updateWordOptions = () => {
```
Comment (msbrogli) [inline src/screens/BackupWords.js]: Maybe rename it to `updateWordsShownOnScreen`.

File: src/screens/BackupWords.js [hunk 5]
```diff
@@ -0,0 +1,241 @@
...
+   * If is the wrong word we show an error and go back to the words screen
+   * If is correct we move one step until the last one.
+   * In case of the last step we show success and redirect to the ChoosePin screen
+   *
+   * @param {String} word Word of the button clicked
+   */
+  wordSelected = (word) => {
+    const index = this.state.indexes[this.state.step];
+    if (this.words[index - 1] === word) {
+      if (this.state.step < 4) {
```
Comment (msbrogli) [inline src/screens/BackupWords.js]: Add a comment: `# Correct word was chosen.`

File: src/screens/BackupWords.js [hunk 6]
```diff
@@ -0,0 +1,241 @@
...
+              onDismiss={() => {
+                this.setState({ modal: null }, () => {
+                  this.props.navigation.navigate('ChoosePinScreen', { words: this.props.navigation.getParam('words') });
+                });
+              }}
+            />,
+        });
+      }
+    } else {
+      // Error
```
Comment (msbrogli) [inline src/screens/BackupWords.js]: Add a comment: `# Incorrect word was chosen.`

File: src/screens/BackupWords.js (hunk 3 above)
Comment (obiyankenobi) [reply src/screens/BackupWords.js]: Yes, mich simpler. Changed it.

File: src/screens/BackupWords.js (hunk 4 above)
Comment (obiyankenobi) [reply src/screens/BackupWords.js]: Done

File: src/screens/BackupWords.js (hunk 5 above)
Comment (obiyankenobi) [reply src/screens/BackupWords.js]: Done

File: src/screens/BackupWords.js (hunk 6 above)
Comment (obiyankenobi) [reply src/screens/BackupWords.js]: Done

File: src/screens/BackupWords.js (hunk 2 above)
Comment (obiyankenobi) [reply src/screens/BackupWords.js]: Yes!

**Pull request merged**